libbeat/statestore/backend/memlog/store.go

```diff
@@ -112,12 +113,21 @@ func openStore(log *logp.Logger, home string, mode os.FileMode, bufSz uint, igno
 		active := dataFiles[L-1]
 		txid = active.txid
 		if err := loadDataFile(active.path, tbl); err != nil {
-			return nil, err
+			if errors.Is(err, ErrCorruptStore) {
+				corruptFilePath := active.path + ".corrupted"
```

Comment:
Is it worth checking that this `.corrupted` file exists at the end of the test?

We must have at least hit this code path for the test to pass at all, so you have proved  we recover which is the important part.